trawler/src/main/java/io/m9sweeper/trawler/ScanRunner.java

```diff
@@ -165,12 +168,21 @@ public void scan() throws Exception {
                                 .data(o.getExtraData())
                 ).collect(Collectors.toList()))).collect(Collectors.toList());
 
-        // If the image hash is different than the one stored in the DB, and it is not a temporary ID, give the scan an error result
-        // Otherwise save the scan results normally
-        if (!scanConfig.getImage().getHash().equals(newImageHash) && (scanConfig.getImage().getHash() != null && !scanConfig.getImage().getHash().startsWith("TMP_"))) {
-            String errorSummary = "Image ID does not match most recent image pulled from Trawler";
-            if (scanConfig.getImage().getTag().contains("latest")) {
-                errorSummary = errorSummary.concat(". Note, using the latest tag for images is not recommended and can cause undesirable behavior");
+        // The hash has changed if the hash from the scan does not match the one from dash AND it is not a placeholder
+        boolean hashHasChanged = !scanConfig.getImage().getHash().equals(newImageHash) && (scanConfig.getImage().getHash() != null && !scanConfig.getImage().getHash().startsWith("TMP_"));
+        boolean compliant = true;
+
+        // If we have an error, or the hash has changed,
+        // Return an error response under the original hash
+        if (hashHasChanged || errorEncountered) {
+            String errorSummary = "";
+            if (hashHasChanged && !errorEncountered) {
```

Comment:
Might be able to simplify logic to...

if (errorEncountered) {
errorSummary = imageTrawlerResultDtos.get(0).getSummary();
} else if (hashHasChanged) {
errorSummary = "Image ID does not match most recent image pulled from Trawler";
                if (scanConfig.getImage().getTag().contains("latest")) {
                    errorSummary = errorSummary.concat(". Note, using the latest tag for images is not recommended and can cause undesirable behavior");
                }
}

I think

Reply:
Though I think what you have will work

Reply:
The if block that chooses the error Summary message is now flipped.